packages/flutter/test/cupertino/popup_surface_test.dart

```diff
@@ -0,0 +1,328 @@
+            isSurfacePainted: false,
+            isVibrancePainted: false,
+            blurSigma: 5,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // Golden displays a CupertinoPopupSurface with no vibrance or surface
+      // color, and a blur sigma of 5
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.blur.5.png'),
+      );
+
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            isSurfacePainted: false,
+            isVibrancePainted: false,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // Golden displays a CupertinoPopupSurface with no vibrance or surface
+      // color, and a blur sigma of 30.
+      await expectLater(
+        find.byType(CupertinoApp),
+        // 30 is the default blur sigma
+        matchesGoldenFile('cupertinoPopupSurface.blur.30.png'),
+      );
+    });
+    testWidgets('Nonpositive blurSigma removes blur', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            isSurfacePainted: false,
+            isVibrancePainted: false,
+            blurSigma: 0,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // The BackdropFilter widget should not be mounted when blurSigma is 0 and
+      // CupertinoPopupSurface.isVibrancePainted is false.
+      expect(
+        find.descendant(
+          of: find.byType(CupertinoPopupSurface),
+          matching: find.byType(BackdropFilter),
+        ),
+        findsNothing,
+      );
+
+      // Golden displays a CupertinoPopupSurface with a blur sigma of 0. Because
+      // the blur sigma == 0 and vibrance and surface are not painted, no popup
+      // surface is displayed.
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.blur.0.png'),
+      );
+
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            isSurfacePainted: false,
+            isVibrancePainted: false,
+            blurSigma: -100,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // Golden displays a CupertinoPopupSurface with a blur sigma of -100. Because
+      // the blur sigma < 0 and vibrance and surface are not painted, no popup
+      // surface is displayed.
+      expect(
+        find.descendant(
+          of: find.byType(CupertinoPopupSurface),
+          matching: find.byType(BackdropFilter),
+        ),
+        findsNothing,
+      );
+
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.blur.0.png'),
+      );
+    });
+  });
+
+  // Because ImageFilter.compose is used to apply multiple filters, the order
+  // of the filters matters. As such, test that the surface effects are stacked
+  // in the correct order.
```

Comment:
Can you explain what this test is for? Also, since composition order matters, can you explain this in the doc? For example, if I want to apply each effect of the pop up surface separately in two `CupertinoPopupSurface`s, what order should I use?

Reply:
I added a clearer doc to tests, while also including how the surface is composed in the CupertinoPopupSurface doc (I assumed this was what you meant by doc, but let me know if I'm wrong). I wasn't sure if you meant I should explicitly say

```dart
/// You can replicate the default appearance of CupertinoPopupSurface with two surfaces: 
Stack(children: <Widget>[
          CupertinoPopupSurface(
            isSurfacePainted: false,
            blurSigma: 0,
            child: SizedBox(),
          ),
          CupertinoPopupSurface(
            isVibrancePainted: false,
            child: SizedBox(),
          )
        ])
```

or if I should just explain composition order, so I kept it simple. Let me know if I should change this.